pinot-common/src/main/java/org/apache/pinot/common/config/provider/TableCache.java

```diff
@@ -142,24 +153,38 @@ public String getActualTableName(String caseInsensitiveTableName) {
    * Returns the table config for the given table, or {@code null} if it does not exist.
    */
   @Nullable
+  @Override
   public TableConfig getTableConfig(String tableNameWithType) {
     return _tableConfigMap.get(tableNameWithType);
   }
 
+  @Override
+  public List<TableConfig> registerTableConfigChangeListener(TableConfigChangeListener tableConfigChangeListener) {
```

Comment:
I guess there will be another PR that includes the code to explicitly invoke these two methods?